src/data-workspace/validation/validation-priority-group-unit.js

```diff
@@ -0,0 +1,85 @@
+import i18n from '@dhis2/d2-i18n'
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { validationLevelsConfig } from './validation-config.js'
+import styles from './validation-priority-group-unit.module.css'
+import {
+    ImportanceLevelPropTypes,
+    ValidationRulePropTypes,
+} from './validation-result-prop-types.js'
+
+const ValidationPriortyGroupUnit = ({ level, validationViolations = [] }) => {
+    if (validationViolations?.length === 0) {
+        return null
+    }
+
+    const validationConfig = validationLevelsConfig[level]
+
+    const style = cx(styles.priorityGroupBox, styles[validationConfig.style])
+
+    const Icon = validationConfig.largeIcon
+    const iconStyle = cx(styles.icon, styles[`${validationConfig.style}Icon`])
+
+    const buildDisplayFormula = (validationRule = {}) => {
+        const { metaData: { leftSide, rightSide, operator } = {} } =
+            validationRule
+
+        // @ToDO: confirm the human readable equivalent for the operators
+        const operatorDisplay = {
+            equal_to: '==',
+            not_equal_to: '!=',
+            greater_than: '>',
+            greater_than_or_equal_to: '>=',
+            less_than: '<',
+            less_than_or_equal_to: '<=',
+            compulsory_pair: 'is compulsory with',
+            exclusive_pair: 'is exclusive to',
+        }
+        return `${leftSide.displayDescription} ${
+            operatorDisplay[operator] ?? operator
+        } ${rightSide.displayDescription}`
+    }
+
+    const translationOptons = {
+        level: validationConfig.text?.toLowerCase(),
+        length: validationViolations.length,
+    }
+
+    return (
+        <div data-test={`priority-group-${level}`}>
+            <div className={styles.titleWrapper}>
+                <div className={iconStyle}>
+                    <Icon />
```

Comment:
Icons have a `color` prop that accepts a string (that is sadly missing from our documentation 😭) that may simplify the styling considerably -- maybe an `iconColor` property could be added to the `validation-config`?